[Bug] The current HEAD of main no longer builds on Rocky Linux 8 & 9.

### Cloudberry Database version

Cloudberrydb main branch (fdc5abc0ca)

### What happened

Compilation error:

```
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-unused-but-set-variable -Werror=implicit-fallthrough=3 -Wno-format-truncation -Wno-stringop-truncation -O3 -fPIC  -DUSE_INTERNAL_FTS=1  -Werror=uninitialized -Werror=implicit-function-declaration -Werror -I../../../../src/interfaces/libpq -I../../../../src/include   -D_GNU_SOURCE -I/usr/include/libxml2   -c -o appendonlyam.o appendonlyam.c
appendonlyam.c: In function ‘appendonly_insert’:
appendonlyam.c:3052:9: error: unused variable ‘firstSequence’ [-Werror=unused-variable]
   int64 firstSequence = GetFastSequences(aoInsertDesc->segrelid,
         ^~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

### What you think should happen instead

This is a relatively new issue. I am able to build at `781c43b63b`.

### How to reproduce

Following the documentation: For RHEL 8 and Rocky Linux 8 (https://cloudberrydb.org/docs/cbdb-linux-compile).

```
./configure --with-perl --with-python --with-libxml --with-gssapi --prefix=/usr/local/cloudberrydb
make
```

### Operating System

Rocky Linux 8 & 9

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://cloudberrydb.org/community/coc).

--- a/src/backend/access/appendonly/appendonlyam.c
+++ b/src/backend/access/appendonly/appendonlyam.c
@@ -3049,12 +3049,13 @@ appendonly_insert(AppendOnlyInsertDesc aoInsertDesc,
 	 */
 	if (aoInsertDesc->numSequences == 0)
 	{
-		int64 firstSequence = GetFastSequences(aoInsertDesc->segrelid,
-											   aoInsertDesc->cur_segno,
-											   aoInsertDesc->lastSequence + 1,
-											   NUM_FAST_SEQUENCES);
+		int64 firstSequence PG_USED_FOR_ASSERTS_ONLY;
 
-		(void)firstSequence;
+		firstSequence = GetFastSequences(aoInsertDesc->segrelid,
+										aoInsertDesc->cur_segno,
+										aoInsertDesc->lastSequence + 1,
+										NUM_FAST_SEQUENCES);
+
 		/* fast sequence could be inconsecutive when insert multiple segfiles */
 		AssertImply(gp_appendonly_insert_files <= 1, firstSequence == aoInsertDesc->lastSequence + 1);
 		aoInsertDesc->numSequences = NUM_FAST_SEQUENCES;

--- a/src/backend/access/appendonly/appendonlyam.c
+++ b/src/backend/access/appendonly/appendonlyam.c
@@ -3052,9 +3052,9 @@ appendonly_insert(AppendOnlyInsertDesc aoInsertDesc,
 		int64 firstSequence PG_USED_FOR_ASSERTS_ONLY;
 
 		firstSequence = GetFastSequences(aoInsertDesc->segrelid,
-										aoInsertDesc->cur_segno,
-										aoInsertDesc->lastSequence + 1,
-										NUM_FAST_SEQUENCES);
+										 aoInsertDesc->cur_segno,
+										 aoInsertDesc->lastSequence + 1,
+										 NUM_FAST_SEQUENCES);
 
 		/* fast sequence could be inconsecutive when insert multiple segfiles */
 		AssertImply(gp_appendonly_insert_files <= 1, firstSequence == aoInsertDesc->lastSequence + 1);

--- a/src/backend/access/appendonly/appendonlyam.c
+++ b/src/backend/access/appendonly/appendonlyam.c
@@ -3054,6 +3054,7 @@ appendonly_insert(AppendOnlyInsertDesc aoInsertDesc,
 											   aoInsertDesc->lastSequence + 1,
 											   NUM_FAST_SEQUENCES);
 
+		(void)firstSequence;
 		/* fast sequence could be inconsecutive when insert multiple segfiles */
 		AssertImply(gp_appendonly_insert_files <= 1, firstSequence == aoInsertDesc->lastSequence + 1);
 		aoInsertDesc->numSequences = NUM_FAST_SEQUENCES;

--- a/src/backend/gpopt/gpdbwrappers.cpp
+++ b/src/backend/gpopt/gpdbwrappers.cpp
@@ -2434,7 +2434,7 @@ gpdb::EvaluateExpr(Expr *expr, Oid result_type, int32 typmod)
 // interpret the value of "With oids" option from a list of defelems
 // GPDB_12_MERGE_FIXME: this leaves dead code in CMDRelationGPDB
 bool
-gpdb::InterpretOidsOption(List *options, bool allowOids)
+gpdb::InterpretOidsOption(List * /*options*/, bool /*allowOids*/)
 {
 	GP_WRAP_START;
 	{
@@ -2549,14 +2549,14 @@ static int64 mdcache_invalidation_counter = 0;
 static int64 last_mdcache_invalidation_counter = 0;
 
 static void
-mdsyscache_invalidation_counter_callback(Datum arg, int cacheid,
-										 uint32 hashvalue)
+mdsyscache_invalidation_counter_callback(Datum /*arg*/, int /*cacheid*/,
+										 uint32 /*hashvalue*/)
 {
 	mdcache_invalidation_counter++;
 }
 
 static void
-mdrelcache_invalidation_counter_callback(Datum arg, Oid relid)
+mdrelcache_invalidation_counter_callback(Datum /*arg*/, Oid /*relid*/)
 {
 	mdcache_invalidation_counter++;
 }

--- a/src/backend/gpopt/translate/CPartPruneStepsBuilder.cpp
+++ b/src/backend/gpopt/translate/CPartPruneStepsBuilder.cpp
@@ -86,7 +86,7 @@ CPartPruneStepsBuilder::CreatePartPruneInfoForOneLevel(CDXLNode *filterNode)
 	{
 		pinfo->subpart_map[i] = -1;
 		if (part_ptr < m_part_indexes->Size() &&
-			i == (int)(*(*m_part_indexes)[part_ptr]))
+			i == (*(*m_part_indexes)[part_ptr]))
 		{
 			// partition did survive pruning
 			pinfo->subplan_map[i] = part_ptr;

--- a/src/backend/gpopt/translate/CQueryMutators.cpp
+++ b/src/backend/gpopt/translate/CQueryMutators.cpp
@@ -1115,7 +1115,7 @@ CQueryMutators::MakeTopLevelTargetEntry(TargetEntry *old_target_entry,
 //		Return the column name of the target list entry
 //---------------------------------------------------------------------------
 CHAR *
-CQueryMutators::GetTargetEntryColName(TargetEntry *target_entry, Query *query)
+CQueryMutators::GetTargetEntryColName(TargetEntry *target_entry, Query * /*query*/)
 {
 	if (nullptr != target_entry->resname)
 	{

--- a/src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp
+++ b/src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp
@@ -531,7 +531,7 @@ CTranslatorDXLToPlStmt::SetParamIds(Plan *plan)
 Plan *
 CTranslatorDXLToPlStmt::TranslateDXLTblScan(
 	const CDXLNode *tbl_scan_dxlnode, CDXLTranslateContext *output_context,
-	CDXLTranslationContextArray *ctxt_translation_prev_siblings)
+	CDXLTranslationContextArray * /*ctxt_translation_prev_siblings*/)
 {
 	// translate table descriptor into a range table entry
 	CDXLPhysicalTableScan *phy_tbl_scan_dxlop =
@@ -976,7 +976,7 @@ CTranslatorDXLToPlStmt::TranslateDXLIndexFilter(
 //---------------------------------------------------------------------------
 void
 CTranslatorDXLToPlStmt::TranslateIndexConditions(
-	CDXLNode *index_cond_list_dxlnode, const CDXLTableDescr *dxl_tbl_descr,
+	CDXLNode *index_cond_list_dxlnode, const CDXLTableDescr * /*dxl_tbl_descr*/,
 	BOOL is_bitmap_index_probe, const IMDIndex *index,
 	const IMDRelation *md_rel, CDXLTranslateContext *output_context,
 	CDXLTranslateContextBaseTable *base_table_context,
@@ -1490,7 +1490,7 @@ CTranslatorDXLToPlStmt::TranslateDXLHashJoin(
 Plan *
 CTranslatorDXLToPlStmt::TranslateDXLTvf(
 	const CDXLNode *tvf_dxlnode, CDXLTranslateContext *output_context,
-	CDXLTranslationContextArray *ctxt_translation_prev_siblings)
+	CDXLTranslationContextArray * /*ctxt_translation_prev_siblings*/)
 {
 	// translation context for column mappings
 	CDXLTranslateContextBaseTable base_table_context(m_mp);
@@ -3788,7 +3788,7 @@ CTranslatorDXLToPlStmt::TranslateDXLCTEProducerToSharedScan(
 Plan *
 CTranslatorDXLToPlStmt::TranslateDXLCTEConsumerToSharedScan(
 	const CDXLNode *cte_consumer_dxlnode, CDXLTranslateContext *output_context,
-	CDXLTranslationContextArray *ctxt_translation_prev_siblings)
+	CDXLTranslationContextArray * /*ctxt_translation_prev_siblings*/)
 {
 	CDXLPhysicalCTEConsumer *cte_consumer_dxlop =
 		CDXLPhysicalCTEConsumer::Cast(cte_consumer_dxlnode->GetOperator());
@@ -5289,7 +5289,7 @@ CTranslatorDXLToPlStmt::TranslateDXLPhyCtasToIntoClause(
 //---------------------------------------------------------------------------
 GpPolicy *
 CTranslatorDXLToPlStmt::TranslateDXLPhyCtasToDistrPolicy(
-	const CDXLPhysicalCTAS *dxlop, List *target_list)
+	const CDXLPhysicalCTAS *dxlop, List * /*target_list*/)
 {
 	ULongPtrArray *distr_col_pos_array = dxlop->GetDistrColPosArray();
 
@@ -5639,7 +5639,7 @@ CTranslatorDXLToPlStmt::TranslateDXLBitmapIndexProbe(
 Plan *
 CTranslatorDXLToPlStmt::TranslateDXLValueScan(
 	const CDXLNode *value_scan_dxlnode, CDXLTranslateContext *output_context,
-	CDXLTranslationContextArray *ctxt_translation_prev_siblings)
+	CDXLTranslationContextArray * /*ctxt_translation_prev_siblings*/)
 {
 	// translation context for column mappings
 	CDXLTranslateContextBaseTable base_table_context(m_mp);

--- a/src/backend/gpopt/translate/CTranslatorQueryToDXL.cpp
+++ b/src/backend/gpopt/translate/CTranslatorQueryToDXL.cpp
@@ -1656,7 +1656,7 @@ CTranslatorQueryToDXL::TranslateWindowSpecToDXL(
 CDXLNode *
 CTranslatorQueryToDXL::TranslateWindowToDXL(
 	CDXLNode *child_dxlnode, List *target_list, List *window_clause,
-	List *sort_clause, IntToUlongMap *sort_col_attno_to_colid_mapping,
+	List * /*sort_clause*/, IntToUlongMap *sort_col_attno_to_colid_mapping,
 	IntToUlongMap *output_attno_to_colid_mapping)
 {
 	if (0 == gpdb::ListLength(window_clause))
@@ -3393,7 +3393,7 @@ CTranslatorQueryToDXL::NoteDistributionPolicyOpclasses(const RangeTblEntry *rte)
 CDXLNode *
 CTranslatorQueryToDXL::TranslateValueScanRTEToDXL(const RangeTblEntry *rte,
 												  ULONG rt_index,
-												  ULONG current_query_level)
+												  ULONG /*current_query_level*/)
 {
 	List *tuples_list = rte->values_lists;
 	GPOS_ASSERT(nullptr != tuples_list);

--- a/src/backend/gpopt/translate/CTranslatorRelcacheToDXL.cpp
+++ b/src/backend/gpopt/translate/CTranslatorRelcacheToDXL.cpp
@@ -744,7 +744,7 @@ CTranslatorRelcacheToDXL::RetrieveRelDistributionOpFamilies(CMemoryPool *mp,
 void
 CTranslatorRelcacheToDXL::AddSystemColumns(CMemoryPool *mp,
 										   CMDColumnArray *mdcol_array,
-										   Relation rel)
+										   Relation /*rel*/)
 {
 	for (INT i = SelfItemPointerAttributeNumber;
 		 i > FirstLowInvalidHeapAttributeNumber; i--)

--- a/src/backend/gpopt/translate/CTranslatorScalarToDXL.cpp
+++ b/src/backend/gpopt/translate/CTranslatorScalarToDXL.cpp
@@ -2055,7 +2055,7 @@ CTranslatorScalarToDXL::TranslateArrayRefToDXL(
 
 CDXLNode *
 CTranslatorScalarToDXL::TranslateSortGroupClauseToDXL(
-	const Expr *expr, const CMappingVarColId *var_colid_mapping)
+	const Expr *expr, const CMappingVarColId * /*var_colid_mapping*/)
 {
 	GPOS_ASSERT(IsA(expr, SortGroupClause));
 	const SortGroupClause *sgc = (SortGroupClause *) expr;

--- a/src/backend/gpopt/translate/CTranslatorUtils.cpp
+++ b/src/backend/gpopt/translate/CTranslatorUtils.cpp
@@ -847,7 +847,7 @@ CTranslatorUtils::GetSystemColType(CMemoryPool *mp, AttrNumber attno)
 
 // Returns the length for the system column with given attno number
 // GPDB_12_MERGE_FIXME: Can we get rid of this function? We should be able to get this info from pg_attribute
-const ULONG
+ULONG
 CTranslatorUtils::GetSystemColLength(AttrNumber attno)
 {
 	GPOS_ASSERT(FirstLowInvalidHeapAttributeNumber < attno && 0 > attno);

--- a/src/include/gpopt/translate/CTranslatorUtils.h
+++ b/src/include/gpopt/translate/CTranslatorUtils.h
@@ -142,7 +142,7 @@ public:
 	static const CWStringConst *GetSystemColName(AttrNumber attno);
 
 	// returns the length for the system column with given attno number
-	static const ULONG GetSystemColLength(AttrNumber attno);
+	static ULONG GetSystemColLength(AttrNumber attno);
 
 	// translate the join type from its GPDB representation into the DXL one
 	static EdxlJoinType ConvertToDXLJoinType(JoinType jt);

